.github/workflows/test-all-deps.yml

```diff
@@ -44,13 +44,7 @@ jobs:
       - run: pip install .
       - run: pip install matplotlib
       - run: pytest --ignore=test/install --cov=fairlearn --cov-report=xml test/
-      - name: Upload coverage to Codecov
-        if: ${{ (inputs.codeCovPython == inputs.python) && (steps.get_date.outputs.AGE < 172800) }} # 172800 = 48 * 3600
-        uses: codecov/codecov-action@v3
-        with:
-          env_vars: OS,PYTHON
-          fail_ci_if_error: true
-          files: ./coverage.xml
-          flags: unittests
-          name: codecov-umbrella
-          verbose: true
+      - name: Upload coverage reports to Codecov with GitHub Action
+        uses: codecov/codecov-action@v4.2.0
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
the token is not really available via secrets on PRs for forks, that's why we have hardcoded it in our config, and it should stay that way, otherwise we quickly encounter GH rate limits and codecove uploads will fail.

Reply:
removed the secrets reference in [73a6af3](https://github.com/fairlearn/fairlearn/pull/1399/commits/73a6af38d8476f5f090d2f6e3d44c5ac15834b87)